From 615b212918e83c4c5161f11596d4c9b49bc1e238 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 03:40:47 +0200
Subject: [PATCH 49/65] acerhk: drop poll kernel parameter

Thanks to DMI based probing, setting this by hand is no longer
required.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -129,13 +129,10 @@ typedef asmlinkage void (*bios_call) (st
 #define XSIZE     20
 
 /* Module parameters */
-static int poll;
 static int verbose = 1;
 
-module_param(poll, int, 0444);
 module_param(verbose, int, 0444);
 
-MODULE_PARM_DESC(poll, "start polling timer");
 MODULE_PARM_DESC(verbose, "output additional information");
 
 /* input device */
@@ -1081,10 +1078,7 @@ static void start_polling(void)
 		add_timer(&acerhk_timer_poll);
 		acerhk_polling_state = 1;
 
-		pr_info("Your hardware does not need polling enabled for hotkeys to work,\n");
-		pr_info("you can safely disable polling by using the module parameter\n");
-		pr_info("poll=0 (unless you want to play around with the driver and see\n");
-		pr_info("if there are buttons which need polling).\n");
+		pr_info("Your hardware does not require polling to be enabled.\n");
 
 		if (verbose)
 			pr_info("starting key polling, every %d ms\n",
@@ -1828,17 +1822,10 @@ static int acerhk_probe(struct platform_
 
 		pr_info("Acer Travelmate hotkey driver successfully loaded.\n");
 
-		/* start automatic polling of key presses if wanted and bios
-		 * routine found */
-		if (poll && bios_routine)
-			start_polling();
 #else
 		setup_model_features();
 		pr_info("Acer Travelmate hotkey driver dummy\n");
 		enable_dritek_keyboard();
-
-		if (poll)
-			start_polling();
 #endif
 		/* enable wlan LED */
 		if (acerhk_model_features & TM_F_WLAN_EC1)
